src/scopeck.rs

```diff
@@ -151,156 +151,148 @@ struct ScopeState<'a> {
     frames_out: Vec<Frame>,
 }
 
-fn push_diagnostic(state: &mut ScopeState, ix: StatementIndex, diag: Diagnostic) {
-    state.diagnostics.entry(ix).or_insert(Vec::new()).push(diag);
-}
+impl<'a> ScopeState<'a> {
+    fn push_diagnostic(&mut self, ix: StatementIndex, diag: Diagnostic) {
```

Comment:
Also mildly annoyed that github's review UI doesn't seem to have a `-b` button, so I'm not sure what all is changed here